src/slotscheck/checks.py

```diff
@@ -48,12 +48,17 @@ def has_duplicate_slots(c: type) -> bool:
 # If the active Python interpreter is the official CPython interpreter,
 # prefer a more reliable CPython-specific solution guaranteed to succeed.
 if platform.python_implementation() == "CPython":
-    # Magic number defined by the Python codebase at "Include/object.h".
+    # Magic numbers defined by the Python codebase at "Include/object.h".
+    Py_TPFLAGS_IMMUTABLETYPE = 1 << 8
     Py_TPFLAGS_HEAPTYPE = 1 << 9
 
     def is_pure_python(cls: type) -> bool:
         "Whether the class is pure-Python or C-based"
-        return bool(cls.__flags__ & Py_TPFLAGS_HEAPTYPE)
+        # Starting with CPython 3.10, `Py_TPFLAGS_HEAPTYPE` should no longer
+        # be relied on, and `!Py_TPFLAGS_IMMUTABLETYPE` should be used instead.
+        return bool(cls.__flags__ & Py_TPFLAGS_HEAPTYPE) and not bool(
```

Comment:
instead of a checking both flags for all python versions, could we instead just define a different `is_pure_python`. That'd improve the performance of this bit of code that is run very often.

```python
if sys.version_info > (3, 10):
    def is_pure_python(cls: type) -> bool:  # the 3.10 version
        ...
else:
    def is_pure_python(cls: type) -> bool:
        ...
```

Reply:
Ah, good point! I wasn't sure if the `Py_TPFLAGS_HEAPTYPE` check would still be needed in 3.10, but looking at the code again it appears that `Py_TPFLAGS_IMMUTABLETYPE` is a full replacement.
Updated in https://github.com/ariebovenberg/slotscheck/pull/104/commits/de44a70643d1bb6932d1e2a6b59edea1c8707bde.